### ISSUE TITLE
`garden publish` command fails because local image is not found

### ISSUE THREAD

**Issue by mkhq** [garden-io/garden#5045]
### Current Behavior

When running `garden publish <target>`, the command fails when trying to upload the image to the remote registry.

```
✖ build.api            → Failed publishing build api (took 15.3 sec)
✖ build.api            → Failed processing publish Build type=container name=api (took 15.33 sec). This is what happened:

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Unable to run docker command: Command ".garden/tools/docker/1cbaf5c26b78806e/docker/docker push <registry>/test/api:latest" failed with code 1:

An image does not exist locally with the tag: <registry>/test/api

Here's the full output:

The push refers to repository [<registry>/test/api]
An image does not exist locally with the tag: <registry>/test/api
```

### Expected behavior

The image should be pushed to the target registry.

### Reproducible example

```
---
kind: Build
type: container
name: api
description: The api service
spec:
  publishId: api:latest
```

### Suggested solution(s)

One option is to make sure we add the missing tag in the local docker repo.

### Your environment

* **OS:** 
* **How I'm running Kubernetes:**  

0.13.13

**Comment by vvagaytsev** [comment]
We have a community PR that might be able to address this: #4074. The tag here seems to contain slashes. That might be the root cause.

**Comment by vvagaytsev** [comment]
Cannot reproduce this with the latest build from the `main` branch. This was likely fixed in #6050 or in one of the earlier releases.
